src/main/java/racingcargame/domain/Cars.java

```diff
@@ -0,0 +1,86 @@
+package racingcargame.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import racingcargame.exception.DuplicateNameException;
+import racingcargame.exception.InvalidNameLengthException;
+
+public class Cars {
+
+	public static final String CARNAME_SPLIT_DELIMITER = ",";
+	public static final int CARNAME_MIN_LENGTH = 1;
+	public static final int CARNAME_MAX_LENGTH = 5;
+	public static final int MAX_MOVEMENT_INIT = 0;
+
+	private List<Car> cars;
+
+	public Cars(String carNames) {
+		if (carNames == null) {
+			throw new NullPointerException();
+		}
+		String[] carNameTokens = carNames.split(CARNAME_SPLIT_DELIMITER);
+
+		validateCarNameLength(carNameTokens);
+		validateDuplicateCarName(carNameTokens);
+
+		List<Car> racingCars = new ArrayList<>();
+		Arrays.stream(carNameTokens).forEach(carName -> racingCars.add(new Car(carName)));
+
+		cars = racingCars;
+	}
+
+	private void validateDuplicateCarName(String[] carNameTokens) {
+		List<String> validTokens = new ArrayList<>();
+
+		Arrays.stream(carNameTokens).forEach(carName -> {
+				if (validTokens.contains(carName)) {
+					throw new DuplicateNameException();
+				}
+				validTokens.add(carName);
+			}
+		);
+	}
+
+	private void validateCarNameLength(String[] carNameTokens) {
+		Arrays.stream(carNameTokens).forEach(carName -> {
+				if (carName.length() < CARNAME_MIN_LENGTH || carName.length() > CARNAME_MAX_LENGTH) {
+					throw new InvalidNameLengthException();
+				}
+			}
+		);
+	}
+
+	public void tryMove(int min, int max) {
+		cars.forEach(car -> {
+				car.tryMove(min, max);
+			}
+		);
+	}
+
+	public List<String> findWinner() {
+		Optional<Car> winnerCar = cars.stream().max(Car::compareTo);
+
+		int maxMovement = MAX_MOVEMENT_INIT;
+		if (winnerCar.isPresent()) {
+			maxMovement = winnerCar.get().getMovement();
+		}
+
+		List<String> winners = new ArrayList<>();
+		for (Car car : cars) {
+			compareMovement(car, maxMovement, winners);
+		}
+		return winners;
+	}
+
+	public void compareMovement(Car car, int max, List<String> winners) {
+		if (car.getMovement() == max) {
```

Comment:
객체에서 데이터를 가져와 어떤 과정을 수행하는 것보다, **객체에 메시징을 해서 응답만 받아내는게** 더 객체지향스러울 것 같아요.

 ```suggestion
		        if (car.hasSameMovement(max)) {
```

제 개인적인 생각으로 비교해봤을때 얻을 수 있는 장점은 다음과 같아요.
1. 만약 그냥 getter를 사용해서 비교하면 movement라는 필드 정보가 수정(타입이 변하는 등)될 경우 이 부분의 수정도 불가피해진다. 그러나 메시징만 할 경우 수정할 필요가 없어진다.
   - 하나의 객체는 다른 객체의 내부 상태를 최대한 모르고 있어야 결합도가 작아지는 것이다.
2. 가독성이 좋아진다. 지금은 단순 하나의 요소 비교 로직이라 큰 차이가 없을 수도 있지만, 긴 로직이라면 하나의 메소드로 빼서 메시징 하는것이 이 코드를 처음 본 사람도 쉽게 이해할 수 있게 될 것이다.  

!!! getter를 아예 쓰지 말자는 의견이 아니니 오해하지 말아주세요 !!!

Reply:
말씀하신 대로 객체에 메시지를 보내야 한다는 것에 동의합니다. 간단한 로직이라 깊게 생각하지 못 한 것 같습니다. 더 객체지향적으로 리팩토링해보겠습니다.